code/mobility-services/mobility-services.js

```diff
@@ -14,63 +28,62 @@ $(document).on('knack-view-render.any', function (event, view, data) {
-/***** Show character limit on paragraph text type ******/
+/********** Show character limit on text type ***********/
 /********************************************************/
 function showCharacterLimit(view_id, field_id, charLimit) {
-  $( document ).ready(function() {
-    const charLeftText = "<p class='typed-chars'>" + charLimit +  " characters allowed</p>";
-    $(".kn-form.kn-view." + view_id + " form #" + field_id).after(charLeftText);
-    $(".kn-form.kn-view."+ view_id + " form #" + field_id).on('input',function(e){
-      const $input = $(this);
-      const charCount = $input.val().length;
-      const totalChar = charLimit - charCount;
-      if (charCount > charLimit) {
-        $input.siblings('.typed-chars').css({"color": "#ff0000", "font-weight": "bold"});
-        $input.siblings('.typed-chars').text(Math.abs(totalChar) + " characters over limit");
-      } else {
-        $input.siblings('.typed-chars').css({"color": "#4a4a4a", "font-weight": "normal"});
-        if (charCount == 0) {
-          $input.siblings('.typed-chars').text(totalChar + " characters allowed");
-        } else {
-          $input.siblings('.typed-chars').text(totalChar + " characters left");
-        }
-      }
+  // Function returns list as message string and css dictionary based on the input length and character limit
+  function showMessage(inputField) {
+    const fieldLength = inputField.val().length;
+    var inputLength = Math.abs(charLimit - fieldLength); // No negative numbers 
+    var fieldText = fieldLength == 0 ? "allowed" : "left";
+    var cssField = {"color": "#4a4a4a", "font-weight": "normal"};
+    if (fieldLength > charLimit) { // if the length is over the character limit change the CSS and text
+      cssField = {"color": "#ff0000", "font-weight": "bold"}; // make text red and bold instead
+      fieldText = "over limit"; // Will say "XXX characters over limit" instead
+    }
+    return [inputLength + " characters " + fieldText,cssField]; // returns list as [str message, dict css]
+  }
+
+  // Shows the message after field input based on character limit and length
+  $(document).on("knack-view-render." + view_id, function(event, view, data) {
+    // When first viewing the field input
+    const formViewFieldID = ".kn-form.kn-view."+ view_id + " form #" + field_id;
+    const fieldMessage = showMessage($(formViewFieldID))[0]; 
+    $(formViewFieldID).after(`<p class='typed-chars'>${fieldMessage}</p>`);
+    
+    // When user is typing in the input field change the text and CSS
+    $(document).ready(function() { 
+      $(formViewFieldID).on('input',function(e){
+        const $input = $(this);
+        const inputMessage = showMessage($input)[0];
+        const cssField = showMessage($input)[1];
+        $input.siblings('.typed-chars').text(inputMessage); // Set text message of field
+        $input.siblings('.typed-chars').css(cssField); // Set CSS of typed-chars class
+      });
     });
   });
-}
+};
```

Comment:
stray closing bracket on line 79?

Reply:
It closes the `function showMessage(inputField)` on line 47